samples/bluetooth/periodic_adv_conn/src/main.c

```diff
@@ -0,0 +1,207 @@
+	.response_slot_spacing = 0x50,
+	.num_response_slots = NUM_RSP_SLOTS,
+};
+
+static struct bt_le_per_adv_subevent_data_params subevent_data_params[NUM_SUBEVENTS];
+static struct net_buf_simple bufs[NUM_SUBEVENTS];
+static uint8_t backing_store[PACKET_SIZE][NUM_SUBEVENTS];
+
+BUILD_ASSERT(ARRAY_SIZE(bufs) == ARRAY_SIZE(subevent_data_params));
+BUILD_ASSERT(ARRAY_SIZE(backing_store) == ARRAY_SIZE(subevent_data_params));
+
+static void request_cb(struct bt_le_ext_adv *adv, const struct bt_le_per_adv_data_request *request)
+{
+	int err;
+	uint8_t to_send;
+
+	/* Continuously send the same dummy data and listen to all response slots */
+
+	to_send = MIN(request->count, ARRAY_SIZE(subevent_data_params));
+	for (size_t i = 0; i < to_send; i++) {
+		subevent_data_params[i].subevent =
+			(request->start + i) % per_adv_params.num_subevents;
+		subevent_data_params[i].response_slot_start = 0;
+		subevent_data_params[i].response_slot_count = NUM_RSP_SLOTS;
+		subevent_data_params[i].data = &bufs[i];
+	}
+
+	err = bt_le_per_adv_set_subevent_data(adv, to_send, subevent_data_params);
+	if (err) {
+		printk("Failed to set subevent data (err %d)\n", err);
+	}
+}
+
+static bool get_address(struct bt_data *data, void *user_data)
+{
+	bt_addr_le_t *addr = user_data;
+
+	if (data->type == BT_DATA_LE_BT_DEVICE_ADDRESS) {
+		memcpy(addr->a.val, data->data, sizeof(addr->a.val));
+		addr->type = data->data[sizeof(addr->a)];
+
+		return false;
+	}
+
+	return true;
+}
+
+static struct bt_conn *default_conn;
+
+static void response_cb(struct bt_le_ext_adv *adv, struct bt_le_per_adv_response_info *info,
+			struct net_buf_simple *buf)
+{
+	int err;
+	bt_addr_le_t peer;
+	char addr_str[BT_ADDR_LE_STR_LEN];
+	struct bt_conn_le_create_synced_param synced_param;
+	struct bt_le_conn_param conn_param;
+
+	if (!buf) {
+		return;
+	}
+
+	if (default_conn) {
+		/* Do not initiate new connections while already connected */
+		return;
+	}
+
+	bt_addr_le_copy(&peer, &bt_addr_le_none);
+	bt_data_parse(buf, get_address, &peer);
+	if (bt_addr_le_eq(&peer, &bt_addr_le_none)) {
+		/* No address found */
+		return;
+	}
+
+	bt_addr_le_to_str(&peer, addr_str, sizeof(addr_str));
+	printk("Connecting to %s in subevent %d\n", addr_str, info->subevent);
+
+	synced_param.peer = &peer;
+	synced_param.subevent = info->subevent;
+
+	/* Choose same interval as PAwR advertiser to avoid scheduling conflicts */
+	conn_param.interval_min = SUBEVENT_INTERVAL;
+	conn_param.interval_max = SUBEVENT_INTERVAL;
+
+	/* Default values */
+	conn_param.latency = 0;
+	conn_param.timeout = 400;
+
+	err = bt_conn_le_create_synced(adv, &synced_param, &conn_param, &default_conn);
+	if (err) {
+		printk("Failed to initiate connection (err %d)", err);
+	}
+}
+
+static const struct bt_le_ext_adv_cb adv_cb = {
+	.pawr_data_request = request_cb,
+	.pawr_response = response_cb,
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	printk("Connected (err 0x%02X)\n", err);
+
+	if (err) {
+		bt_conn_unref(conn);
```

Comment:
You don’t own the conn reference here, so you can’t unref it. If you stored a reference in some local variable (like default_conn) then sure, go ahead and unref that (and set it to NULL), but conn here belongs to the stack.

Reply:
Updated to operate on `default_conn`. `bt_conn_le_create_synced` sets `default_conn` and gives a reference so that needs to be unrefed here.